src/tree/structure/TextContainer.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Represents a text (with optional formatting element(s)) within a document that can be read by a reader.
+ *
+ * Example (in the case of HTML):
+ * ```html
+ * This text is <strong id="elem-id">very strong</strong>.
+ * ```
+ * is transformed to:
+ * ```js
+ * TextContainer {
+ *     text: "This text is very strong".
+ *     formatting: [
+ *         FormattingElement {
+ *             type: "strong",
+ *             startIndex: 13,	// "This text is ".length
+ *             endIndex: 54, 	// "This text is <strong id="elem-id">very strong</strong>".length
+ *             startText: 13, 	// "This text is ".length
+ *             endText: 24,	// "This text is very strong".length
+ *             attributes: {
+ *                 id: "elem-id"
+ *             }
+ *         }
+ *     ]
+ * }
+ * ```
+ *
+ * @memberOf module:tree/structure
```

Comment:
Unfortunately JSDoc does not seem to play nice with this approach 🙁.
Having multiple `@module` comments in multiple files leads to multiple module definitions, instead of merging them into one.

See also: https://github.com/jsdoc3/jsdoc/issues/1002